File: techila/examples/R/Features/custom_library/run_packagetest2.r
# Copyright 2010-2013 Techila Technologies Ltd.

# This R-script contains the Local Control Code.
#
# Load the techila library
library(techila)

# This function will call the 'peach' function, which will create the
# Project.
#
# Usage: result <- run_packagetest2(input)
# input = String
# Example:
# result <- run_packagetest2("testvalue")
run_packagetest2 <- function(input) {

  result <- peach(funcname = "packagetest_dist", # Function that will be executed on Workers
                  params = list(input), # Input parameters for the executable function
                  files = list("packagetest_dist.r"), # Files that will be sourced on Workers
                  packages = list("techilaTestPackage"),
                  peachvector = 1:1, # Set the number of Jobs to one (1)
                  sdkroot = "../../../.." # Location of the techila_settings.ini file
                 )
}

File: techila/examples/python/Features/custom_package/run_packagetest2.py
# Copyright 2012-2016 Techila Technologies Ltd.

# This script contains the Local Control Code containing function
# definition for running packagetest example.
#
# Usage:
#
# execfile('run_packagetest2.py')
# result = run_packagetest2()
#
# Import the techila package
import techila

# This function is used to create a computational Project that will import
# the Bundle containing the 'techilatest' package.
def run_packagetest2():
    results = techila.peach(funcname = 'packagetest_dist', # Function executed on Workers
                            params = ['<vecidx>','<param>'], # Input arguments to the executable function
                            files = ['packagetest_dist.py'], # File that will be evaluated at the start
                            packages = ['techilatest'],
                            peachvector = [1, 2, 4, 8, 16], # Peachvector containing five integer elements
                            )

    # Display the results
    for row in results:
        print(row)
